main/json.c

```diff
@@ -861,6 +861,83 @@ enum ast_json_to_ast_vars_code ast_json_to_ast_variables(struct ast_json *json_v
 	return AST_JSON_TO_AST_VARS_CODE_SUCCESS;
 }
 
+enum ast_json_nvp_ast_vars_code ast_json_nvp_array_to_ast_variables(
+	struct ast_json *json_variables, struct ast_variable **variables)
+{
+	struct ast_variable *tail = NULL;
+	int i = 0;
+	size_t len = json_variables ? ast_json_array_size(json_variables) : 0;
+
```

Comment:
Should *variables be set to NULL at the start?

Reply:
No because there may already be variables you want to keep. For this case, they get accumulated from the query string, the query_strings property and the the message body.  I'll document that.  Actually, thinking about it, a new variable should replace an existing one with the same name.  I'll fix that as well.

Reply:
done